sklearn/utils/_print_versions.py

```diff
@@ -0,0 +1,156 @@
+"""
+Utility methods to print system info for debugging
+
+adapted from :func:`pandas.show_versions`
+"""
+# License: BSD 3 clause
+
+import os
+import platform
+import sys
+import struct
+import subprocess
+import codecs
+import locale
+import importlib
+
+from .._build_utils import get_blas_info
+
+
+def get_sys_info():
+    """System information
+
+    Return
+    ------
+    sys_info : dict
+        system and Python version information
+
+    """
+
+    blob = []
+
+    # get full commit hash
+    commit = None
+    if os.path.isdir(".git") and os.path.isdir("sklearn"):
```

Comment:
This code branch is currently not tested. I'm not sure we actually want it. It would be nice, not sure we should bother with it. @lesteve ?

Reply:
I don't think it is really useful to include the git commit info. In my mind this is useful for users that come with a bug report. In 99.9% of the use cases they are using a released version.

In the case they are not using a released version, then the person can very easily provide this info.